### PR TITLE
Cucumber Tests

### PR DESCRIPTION
**Pull request by xenown**

### CODE REVIEW

File: src/__tests__/cucumber_tests/shared_steps/equalizerApo.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  given('EqualizerAPO is installed', async () => {
+    if (!(await isEqualizerAPOInstalled())) {
+      throw new Error('EqualizerAPO not installed');
+    }
+    // TODO find a way to install EqualizerAPO
+  });
+};
+
+export const givenCanWriteToAquaConfig = (given: DefineStepFunction) => {
+  given('Aqua can write to Aqua config', async () => {
```
Comment (selinali2010) [inline src/__tests__/cucumber_tests/shared_steps/equalizerApo.ts:15]: this is out of scope for this PR but would we be able to leverage this given to manually set the initial state of the equalizer?

File: src/__tests__/cucumber_tests/shared_steps/aquaSlider.ts [hunk 2]
```diff
@@ -1,46 +1,362 @@
...
+
+const setPreampGainNumber = async (
+  webdriver: { driver: Driver | undefined },
+  gain: number
+) => {
+  const inputElement = await webdriver.driver.$(
+    '.sideBar input[name="Pre-Amplification Gain (dB)-number"]'
+  );
+  await inputElement.setValue(gain);
+  await inputElement.keys('Tab');
```
Comment (selinali2010) [inline src/__tests__/cucumber_tests/shared_steps/aquaSlider.ts:318]: Would simulating a key press of `Enter` make more intuitive sense? I'm fine with leaving it as `Tab` as well

File: src/__tests__/cucumber_tests/shared_steps/aquaSlider.ts [hunk 3]
```diff
@@ -1,46 +1,362 @@
...
+      );
+      expect(filterElement).not.toBeNull();
+      filterElement.click();
+      // wait 1000 ms for the action.
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+    }
+  );
+};
+
+export const whenSetFrequencyFilterType = (
```
Comment (selinali2010) [inline src/__tests__/cucumber_tests/shared_steps/aquaSlider.ts:195]: We can refactor the code in this function and `givenFrequencyFilterType` into a `setFrequencyFilterType` function similar to how you handled the other test cases. It's probably more readable that way

File: src/__tests__/cucumber_tests/shared_steps/aquaSlider.ts [hunk 4]
```diff
@@ -1,46 +1,362 @@
...
+
+// ====================================
+
+const setFrequencyQuality = async (
+  webdriver: { driver: Driver | undefined },
+  frequency: number,
+  quality: string
+) => {
+  const element = await webdriver.driver.$(
+    `.mainContent label[for="${frequency}-quality"]`
```
Comment (selinali2010) [inline src/__tests__/cucumber_tests/shared_steps/aquaSlider.ts:106]: is there a reason why we can't just do `.mainContent label[for="${frequency}-quality"] input` directly?

**Review by selinali2010 — APPROVED** [review]
Minor comments, but otherwise LGTM!

File: src/__tests__/cucumber_tests/shared_steps/aquaSlider.ts (hunk 2 above)
Comment (xenown) [reply src/__tests__/cucumber_tests/shared_steps/aquaSlider.ts:318]: It didn't work with Enter :')

**Pull request merged**